package taggit_pb;

message User {
       optional int32 id = 1;
       optional string name = 2;
}

message Tag {
       optional int32 id = 1;
       optional string name = 2;
}

message Item {
	optional int32 id = 1;
	optional string value = 2;
	extensions 100 to max;
}

message Paper {
	extend Item {
               optional Paper paper = 101;
	}  
	optional string title = 1;
	optional string author = 2;
	optional string year = 3;
	optional string url = 4;
	optional string lastmod = 5;
}

message UserTagItem {
	optional int32 user_id = 1;
	optional int32 tag_id = 2;
	optional int32 item_id = 3;
}

message VLists {
	repeated User users = 1;
	repeated Tag  tags  = 2;
	repeated Item items = 3;
	repeated UserTagItem utis = 4;
	repeated Paper papers= 5;   

	optional string query = 11;
	optional int32 query_offset = 12 [default = 0];
	optional int32 query_size = 13 [default = 0];
	
        optional string status = 101;
}

message JsonRequest {
	optional VLists input = 1;
}

message JsonResponse {
	optional VLists output = 1;
}
